### PR TITLE
Bump chalk to use latest git to get upstream fix

### PR DESCRIPTION
**Pull request by Ameobea**
* Chalk very recently (like an hour ago) merged a fix that prevents rust analyzer from panicking.  This allows it to be usable again for code that hits those situations.  See #6134, #6145, Probably #6120

### CODE REVIEW

File: crates/hir_ty/src/traits/chalk/mapping.rs [hunk 1]
```diff
@@ -399,6 +399,7 @@ impl ToChalk for TypeCtor {
                 // this should not be reached, since we don't represent TypeName::Error with TypeCtor
                 unreachable!()
             }
+            _ => unimplemented!(), // FIXME
```
Comment (lnicola) [inline crates/hir_ty/src/traits/chalk/mapping.rs:402]: It might be better to name all the variants here, so we know what we're missing.

Comment (Ameobea) [reply crates/hir_ty/src/traits/chalk/mapping.rs]: Good call - implemented

**Comment by lnicola** [comment]
Hm, I wonder if we could get the `chalk` folks to tag a release.

**Comment by lnicola** [comment]
Can you update to the latest chalk release?

**Comment by Ameobea** [comment]
Updated

**Review by lnicola — APPROVED** [review]

**Pull request merged**